sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala

```diff
@@ -532,7 +544,8 @@ object CatalogColumnStat extends Logging {
         nullCount = map.get(s"${colName}.${KEY_NULL_COUNT}").map(v => BigInt(v.toLong)),
         avgLen = map.get(s"${colName}.${KEY_AVG_LEN}").map(_.toLong),
         maxLen = map.get(s"${colName}.${KEY_MAX_LEN}").map(_.toLong),
-        histogram = map.get(s"${colName}.${KEY_HISTOGRAM}").map(HistogramSerializer.deserialize)
+        histogram = map.get(s"${colName}.${KEY_HISTOGRAM}").map(HistogramSerializer.deserialize),
+        version = map.get(s"${colName}.${KEY_VERSION}").map(_.toInt)
```

Comment:
The version must exist in the table property. I think we can fail earlier if the version doesn't exist, which means the table property is malformed.

Reply:
Do you mean to fail here and replace the type of the `version` field in `CatalogColumnStat` by `Int` (from `Option[Int]`).

Reply:
Yea. If the `version` doesn't exist, we can't just assume the table properties was written by the current Spark version.

Reply:
done